Scripts/RemediationScripts/Remediate-InvalidAADObjectRoleAssignments.ps1

```diff
@@ -83,15 +83,15 @@ function Pre_requisites
     if($availableModules.Name -notcontains 'Az.ResourceGraph')
     {
         Write-Host "Installing module Az.ResourceGraph..." -ForegroundColor Yellow
-        Install-Module -Name AzureAD -Scope CurrentUser -Repository 'PSGallery'
+        Install-Module -Name Az.ResourceGraph -Scope CurrentUser -Repository 'PSGallery'
     }
     else
     {
         Write-Host "Az.ResourceGraph module is available." -ForegroundColor Green
     }
 
-    # Checking if 'AzureAD' module is available or not.
-    if($availableModules.Name -notcontains 'AzureAD')
+    # Checking if 'AzureAD' or 'AzureADPreview' module is available or not.
+    if(($availableModules.Name -notcontains 'AzureAD') -and ($availableModules.Name -notcontains 'AzureADPreview'))
```

Comment:
Is any command being used which might be present in AzureAd but not in preview package

Reply:
Yes, a few commands, such as Get-AzureADTenantDetail and Get-AzureADObjectByObjectId are being used in this script which are available in the AzureAD module but not in the preview package.

To resolve this, we can force the installation of the AzureAD module, which will allow us to install it even if the preview module exists. Additionally, using the -Prefix parameter when importing the module can help prevent confusion between commands that are common to both modules.

Reply:
So above code needs to be modified to support the installation of the AzureAD module even if preview is available

Reply:
FYI, both the commands you mentioned are present in AzureADPreview

https://learn.microsoft.com/en-us/powershell/module/AzureAD/get-azureadtenantdetail?view=azureadps-2.0-preview

https://learn.microsoft.com/en-us/powershell/module/azuread/get-azureadobjectbyobjectid?view=azureadps-2.0-preview